### ISSUE TITLE
add truffle test that withdraw doesn't allow reentry (dao bug)

### ISSUE THREAD

**Issue by snd** [paritytech/parity-bridge#122]
to ensure we never regress on this

1. prepare message and signatures that would be accepted by `withdraw`
2. deploy `HackAttempt` contract that has fallback function that calls `HomeBridge(msg.sender).withdraw` with that message and signatures
  - limit reentry so it doesn't fail by running out of gas
3. deploy `HomeBridge`
4. call `HomeBridge.withdraw` with message and signatures
5. assert that value got transferred only once

comment out `require(!withdraws[hash]);` and confirm test fails then

**Comment by tomusdrw** [comment]
Contracts are not part of this repo any more, so closing.
